### PR TITLE
ci(circle) add build & build-dev jobs

### PR DESCRIPTION
**Pull request by jmaupetit**
NB: The build-dev job depends on the build job to speed up build time by using production container cache.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+          key: edx-v1-{{ .Revision }}
+      - run:
+          name: Save container
+          command: |
+            mkdir ~/docker
+            docker save edxapp | gzip > ~/docker/edxapp.tgz
+      - save_cache:
+          paths:
+            - ~/docker
+          key: docker-v1-{{ .Revision }}
```
Comment (sampaccoud) [inline .circleci/config.yml]: Didn't get the logic in the name of each step?

File: .circleci/config.yml [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+
+  hub:
+    machine: true
+    steps:
+      - restore_cache:
+          keys:
+            - docker-v1-{{ .Revision }}
+      - restore_cache:
+          keys:
+            - docker-dev-v1-{{ .Revision }}
```
Comment (sampaccoud) [inline .circleci/config.yml]: why do you need the dev cache for the production image ?

File: .circleci/config.yml [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+            - build-dev
+          filters:
+            tags:
+              only: /.*/
+      - test:
+          requires:
+            - build-dev
+          filters:
+            tags:
+              only: /.*/
```
Comment (sampaccoud) [inline .circleci/config.yml]: ok, I would need much more comments to understand this file. Maybe someone who knows circleCI can review.

File: .circleci/config.yml [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+            bin/clone_repositories
+      - run:
+          name: Build container
+          command: |
+            docker build -t edxapp:latest .
+      - run:
+          name: Tag container
+          command: |
+            docker tag edxapp fundocker/edxapp:ginkgo.1
+            docker images edxapp
```
[the remark this answers is not in the document]
Comment (jmaupetit) [reply .circleci/config.yml]: It stands for logging purpose, I want to check tags during the build.

File: .circleci/config.yml (hunk 1 above)
Comment (jmaupetit) [reply .circleci/config.yml]: Do you mean step names or cache keys?

File: .circleci/config.yml (hunk 2 above)
Comment (jmaupetit) [reply .circleci/config.yml]: I need to load both production and development images to tag them and push them to docker hub.

File: .circleci/config.yml (hunk 3 above)
Comment (jmaupetit) [reply .circleci/config.yml]: What can I do to improve this?

File: .circleci/config.yml (hunk 1 above)
Comment (sampaccoud) [reply .circleci/config.yml]: I meant step names. The fact that there are duplicates.
I think I would find it nice that each step has a comment line (I mean Ansible style where the name of each task is long enough to explain what it does)

File: .circleci/config.yml (hunk 2 above)
Comment (sampaccoud) [reply .circleci/config.yml]: I couldn't think of a reason to push the dev image should go to docker hub. WDYT ?

File: .circleci/config.yml (hunk 3 above)
Comment (sampaccoud) [reply .circleci/config.yml]: see above. One comment per step ?

File: .circleci/config.yml (hunk 1 above)
Comment (jmaupetit) [reply .circleci/config.yml]: CircleCI configuration file format does not allow custom names for built-in steps such as `save_cache`. But as suggested below, I can add comments.

File: .circleci/config.yml (hunk 2 above)
Comment (jmaupetit) [reply .circleci/config.yml]: This is the image you'll need to develop an Open edX application such as Fonzie (with default configurations and installed dependencies).

File: .circleci/config.yml (hunk 3 above)
Comment (jmaupetit) [reply .circleci/config.yml]: Good idea.

File: .circleci/config.yml [hunk 5]
```diff
@@ -0,0 +1,262 @@
...
+      # This script requires CI environment variable to run; those are passed to
+      # the container via the `-e` docker run option:
+      #
+      # - CIRCLECI: used by the CI script to select the CI strategy (depending
+      #   on the CI platform)
+      # - CIRCLE_NODE_INDEX/CIRCLE_NODE_TOTAL: used for the parallel execution
+      #   of the test suite
+      # - EDXAPP_TEST_MONGO_HOST: the mongo db host used in tests
+      #
+      # We also need to mount the git repository that is used to inegrate
```
Comment (sampaccoud) [inline .circleci/config.yml]: s/inegrate/integrate

File: .circleci/config.yml [hunk 6]
```diff
@@ -0,0 +1,262 @@
...
+
+  # We have a single workflow
+  edxapp:
+
+    jobs:
+
+      # The build job has no required jobs, hence this will be our first job
+      - build:
+          # Filtering rule to run this job: none (we accept all tags; this job
+          # will always run).
```
Comment (sampaccoud) [inline .circleci/config.yml]: :1st_place_medal: +1:

**Pull request merged**